roles/validations/tasks/main.yml

```diff
@@ -0,0 +1,46 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+# We can execute all defined validations when cifmw_validation_run_all is defined.
+# Else, we will skip this and run only the explicitly defined validations from
+# cifmw_validations_list
+- name: Run all validations if required
+  when: cifmw_validations_run_all | bool
+  block:
+    - name: Find all validations
+      register: found_validations
+      ansible.builtin.find:
+        path: "{{ role_path }}/tasks"
+        pattern: .yml
+        recursive: true
+
+    - name: Print found validations
+      ansible.builtin.debug: "Found validations: {{ found_validations }}"
+
+    - name: Run all found validations
+      ansible.builtin.include_tasks: "{{ item }}"
+      loop: "{{ found_validations }}"
+      when: found_validations is exist
+
+- name: Assert all listed validations exist
```

Comment:
I'd make a block with this assert and the include_tasks, in order to apply a `when: not cifmw_validation_run_all | bool` to both. It should be exclusive in a `list XOR all` way IMHO